### PR TITLE
libsndfile 1.0.29pre2

### LINKED ISSUE #57833: libsndfile 1.0.28 contains multiple security bugs
libsndfile 1.0.28 contains multiple security bugs. See <https://www.cvedetails.com/vulnerability-list/vendor_id-16294/product_id-36889/Libsndfile-Project-Libsndfile.html>. Amongst others, also CVE-2017-12562, which causes a crash in openmpt123 (<https://lib.openmpt.org/>) when rendering to wav files. See <https://bugs.openmpt.org/view.php?id=974> and <https://github.com/erikd/libsndfile/issues/292>.

Please update to at least [libsndfile 1.0.29-pre2](http://www.mega-nerd.com/libsndfile/files/1.0.29pre2/libsndfile-1.0.29pre2.tar.bz2) or get the fixes for this CVE (and others) from [git master](https://github.com/erikd/libsndfile/tree/master). See https://github.com/erikd/libsndfile/issues/470 for further discussion.

Other distributions (like e.g. Debian (<https://security-tracker.debian.org/tracker/source-package/libsndfile>)) have already fixed these issues.

### PR DESCRIPTION
**Pull request by manxorist**
Closes #57833.

### CODE REVIEW

**Comment by SMillerDev** [comment]
Could you add a comment that this is because it patches security issues. Normally we don't allow pre releases but there's a pretty good reason here.

File: Formula/libsndfile.rb [hunk 1]
```diff
@@ -1,8 +1,8 @@
 class Libsndfile < Formula
   desc "C library for files containing sampled sound"
   homepage "http://www.mega-nerd.com/libsndfile/"
-  url "http://www.mega-nerd.com/libsndfile/files/libsndfile-1.0.28.tar.gz"
-  sha256 "1ff33929f042fa333aed1e8923aa628c3ee9e1eb85512686c55092d1e5a9dfa9"
+  url "http://www.mega-nerd.com/libsndfile/files/1.0.29pre2/libsndfile-1.0.29pre2.tar.bz2"
```
Comment (claui) [inline Formula/libsndfile.rb:4]:
```suggestion
  # Change back to stable when upstream releases > 1.0.28
  # Fixes several security vulnerabilities. See e. g. PR 408, 432, 439
  url "http://www.mega-nerd.com/libsndfile/files/1.0.29pre2/libsndfile-1.0.29pre2.tar.bz2"
```

**Review by claui — CHANGES_REQUESTED** [review]

File: Formula/libsndfile.rb [hunk 2]
```diff
@@ -1,6 +1,24 @@
...
+  #  CVE-2017-14245
+  #  CVE-2017-14246
+  #  CVE-2017-14634
+  #  CVE-2018-13139
+  #  CVE-2018-13419
+  #  CVE-2018-19432
+  #  CVE-2018-19661
+  #  CVE-2018-19662
+  #  CVE-2018-19758
+  #  CVE-2019-3832
```
Comment (claui) [inline Formula/libsndfile.rb:21]:
```suggestion
  # Fixes several security vulnerabilities, for example:
  # CVE-2017-6892, CVE-2017-12562, CVE-2018-13139
```

I don’t see how enumerating every single related CVE helps, other than to prove a point – which is not necessary as we’re already considering your submission, and given that upstream has the fixes merged to `master` already.

CVE-2017-6892, CVE-2017-12562 and CVE-2018-13139 are [the three most critical ones](https://www.cvedetails.com/vulnerability-list.php?vendor_id=16294&product_id=36889&version_id=213074&page=1&cvssscoremin=6).

**Review by claui — CHANGES_REQUESTED** [review]

**Review by claui — APPROVED** [review]

**Comment by claui** [comment]
Thanks @manxorist for taking care of our users’ security 🍻

**Pull request closed without merging**